csfieldguide/static/interactives/bias-in-big-data/js/bias-in-big-data.js

```diff
@@ -0,0 +1,159 @@
+});
+
+
+function init() {
+    // get random background colour to start
+    startColour =  getRandomColour();
+    sliderStartPos = SLIDER_COLOUR_VALUES[startColour];
+    $('#circles-area').addClass(startColour);
+    // make sure we have at least 3 circles that are the same as the background colour
+    createCircle(startColour);
+    createCircle(startColour);
+    createCircle(startColour);
+    // generate 8 more randomly coloured circles
+    for (i=0; i < NUM_CIRCLES_TO_ADD; i++) {
+        createCircle();
+    }
+    $('.circle').removeClass('d-none');
+    $('.circle').click(function() {
+        // add glow around clicked circle
+        $(this).addClass('glow');
+    });
+    $('#next-stage').removeClass('d-none');
+    $('#start-again').addClass('d-none');
+    createSlider();
+    bgColourSlider[0].noUiSlider.on('update', updateSlider);
+}
+
+
+function createSlider() {
+    noUiSlider.create(bgColourSlider[0], {
+        start: sliderStartPos,
+        step: 1,
+        connect: "lower",
+        orientation: "horizontal",
+        range: {
+          'min': SLIDER_MIN,
+          'max': SLIDER_MAX
+        },
+        format: wNumb({
+          decimals: 0
+        })
+    });
+}
+
+
+function updateSlider() {
+    var value = bgColourSlider[0].noUiSlider.get();
+    hslColour = 'hsl(' + value + ', 100%, 50%)';
+    $('#circles-area').css('background', hslColour);
+    $('.noUi-handle').css('background', hslColour);
+}
+
+
+function getRandomPosition() {
+    var circlesAreaHeight = $('#circles-area').height();
+    var circlesAreaWidth = $('#circles-area').width();
+    var randHeight = Math.floor((Math.random() * circlesAreaHeight));
+    var randWidth = Math.floor((Math.random() * circlesAreaWidth));
+
+    // convert px to %
+    heightInPercentage = Math.floor((randHeight / circlesAreaHeight) * 100);
+    widthInPercentage = Math.floor((randWidth / circlesAreaWidth) * 100);
+
+    // reduces or increases percentage by 5% to prevent circles going outside of parent
+    if (heightInPercentage >= PERENTAGE_BOUNDARY_UPPER) {
+        heightInPercentage -= PERCENTAGE_ADJUSTMENT;
+    } else if (heightInPercentage <= PERENTAGE_BOUNDARY_LOWER) {
+        heightInPercentage += PERCENTAGE_ADJUSTMENT
+    }
+    if (widthInPercentage >= PERENTAGE_BOUNDARY_UPPER) {
+        widthInPercentage -= PERCENTAGE_ADJUSTMENT;
+    } else if (widthInPercentage <= PERENTAGE_BOUNDARY_LOWER) {
+        widthInPercentage += PERCENTAGE_ADJUSTMENT;
+    }
+
+    return {
+        top: heightInPercentage + '%',
+        left: widthInPercentage + '%'
+    };
+}
+
+
+function createCircle(colour) {
+    var colour =  colour || getRandomColour();
+    var $circle = $("<div>").addClass('circle dynamic d-none ' + colour);
+    $circle.css(getRandomPosition());
+    $('#circles-area').append($circle);
+}
+
+
+function loadNextStage() {
+    if (firstStage) {
+        // what happens if they haven't missed any circles?
```

Comment:
It's a bit cheaty but maybe add one more circle afterwards so they definitely miss one? Data being excluded is a bias of course :p

Reply:
Speaking of other biases, did you consider the the zoom-out functionality proposed by the previous stakeholders

Reply:
Decided not to implement because of the following sentence:

> According to the beta feedback from our client, however, the current version was short
enough to hold users’ attention and successfully address the learning outcome.